### PR TITLE
Adds Front-End iOS Practice docs

### PR DESCRIPTION
**Pull request by ashfurrow**
This adds docs for the Front-End iOS Practice, addressing the biggest question: why do we have a separate iOS practice? I followed the format of the [Front-End Practice docs](https://github.com/artsy/README/blob/master/practices/front-end.md) and added what made sense for us.

I'm going to leave this PR unassigned and leave it open for a while, RFC-style. Take your time providing feedback.

### CODE REVIEW

File: practices/front-end-ios.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+Initially, Artsy apps were built with Objective-C. Then Swift was announced in 2014, and
+[we tried it](http://artsy.github.io/blog/2017/02/05/Retrospective-Swift-at-Artsy/) but have ultimately
+[settled on React Native](http://artsy.github.io/blog/2018/03/17/two-years-of-react-native/). We will still use
+Objective-C and Swift for the foreseeable future, and native code will likely always make sense for parts of our
+app (notably, [the Augmented Reality feature](http://artsy.github.io/blog/2018/03/18/ar/) and the animation-driven
+[Live Auctions Integration](http://artsy.github.io/blog/2016/08/09/the-tech-behind-live-auction-integration/#The.iOS.native.app:.Eigen),
+user interface). New features are being built using React Native; the remaining native parts of our main app are
+[listed here 🔒](https://www.notion.so/artsy/Eigen-migration-to-React-Native-54dda83b023b4cb4965a8defdae9687f) with
+a decisions and priorities for switching them to React Native. We have no plans at this time to rewrite our
```
Comment (orta) [inline practices/front-end-ios.md]: > with a decisions and priorities for switching them to React Native

Grammar

File: practices/front-end-ios.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Our goals are the same as the [Front-End Practice goals](./front-end.md#team-goals), but with a focus on supporting
+iOS software at Artsy. That means we have the following additional goals:
+
+- Build tools and systems to help non-iOS engineers contribute to Artsy iOS software using React Native.
+- Help our colleagues understand the differences between building iOS software and web software.
+
+### Logistics
+
+Each iOS app at Artsy is deployed according to its own unique needs. See their repos for deployment instructions.
```
Comment (orta) [inline practices/front-end-ios.md]: Could probably do with a link to the apps below

File: practices/front-end-ios.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+[sometimes in native code](http://artsy.github.io/blog/2018/06/15/cocoapods-keys-react-native/)).
+
+## Team Information
+
+There are two important Slack groups:
+
+- **@ios-front-enders** Engineers who work on Artsy's iOS codebases in a native capacity.
+- **@app-release-team** The
+  [iOS App Quality Team 🔒](https://www.notion.so/artsy/Artsy-iOS-App-Quality-Team-58b9b55f52484000b9689edd809e7016)
+  is a team of one each Designer, Product Manager, and Engineer who help ensure the
```
Comment (orta) [inline practices/front-end-ios.md]: > is a team of one each Designer

feels a bit awkward

File: practices/front-end-ios.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  production; we cut betas and submit to Apple for App Store approval).
+- Long-term maintenance is different (Front-End and Platform practices deploy code to hardware they control and can
+  patch easily; we deploy code to hardware our users control and can't guarantee upgrading at all, so we have to
+  anticipate how iOS code will continue to operate in production for _years_).
+
+We do have a lot in common with the [Front-End Practice](./front-end.md):
+
+- Large parts of our codebase are written in the [Artsy Omakase](./front-end.md#the-artsy-omakase).
+- Our React/Native apps share [a common design system](https://github.com/artsy/palette/).
+- We're all solving problems for Artsy.
```
Comment (orta) [inline practices/front-end-ios.md]: 10/10

File: practices/front-end-ios.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Artsy has the following iOS applications, which are all open source.
+
+- [Eigen](https://github.com/artsy/eigen) is Artsy's main iOS app. (Objective-C, Swift)
+  - [Emission](https://github.com/artsy/emission) is this app's React Native component library (see
+    [this post](http://artsy.github.io/blog/2018/04/17/making-a-components-pod/)). (TypeScript)
+- [Energy](https://github.com/artsy/energy) is Artsy's partner app. (Objective-C)
+- [Eidolon](https://github.com/artsy/eidolon) is Artsy's auctions kiosk. (Swift)
+- [Emergence](https://github.com/artsy/emergence) is Artsy's Apple TV app. (Swift)
+
+All Artsy iOS codebases start with the letter `e`, but not all Artsy codebases that start with the letter `e` are
```
Comment (zephraph) [inline practices/front-end-ios.md]: Oh, that's fascinating. I wasn't aware of this! Now that I think about it though...

**Comment by zephraph** [comment]
Looks great!

**Comment by ashfurrow** [comment]
Cool, let's merge 👍

**Pull request merged**